src/Perlang.Tests.Integration/IndexOperator/StringIndexing.cs

```diff
@@ -7,44 +7,61 @@ namespace Perlang.Tests.Integration.IndexOperator;
 public class StringIndexing
 {
     [Fact]
-    public void AsciiString_can_be_indexed_by_integer()
+    public void ASCIIString_can_be_indexed_by_integer()
     {
-        string source = @"
-            print ""foobar""[0];
-        ";
+        string source = """
+            print "foobar"[0];
+            """;
 
         var output = EvalReturningOutputString(source);
 
         output.Should().Be("f");
     }
 
-    [SkippableFact]
-    public void AsciiString_indexed_outside_string_returns_expected_error()
+    [Fact]
+    public void ASCIIString_indexed_outside_string_returns_expected_error()
     {
-        // -Warray-bounds is not enough to catch this, since the Perlang code is approximately
-        // perlang::ASCIIString::from_static_string("foobar")->char_at(10). We need to implement this on the Perlang side.
-        Skip.If(PerlangMode.ExperimentalCompilation, "String bounds checking is not yet supported in compiled mode");
+        string source = """
+            print "foobar"[10];
+            """;
+
+        var result = EvalWithRuntimeErrorCatch(source);
+
+        result.Errors.Should()
+            .ContainSingle()
+            .Which
+            .Message.Should().Contain("exited with exit code 134");
+
+        result.OutputAsString.Should()
+            .Contain("Index 10 is out-of-bounds for a string with length 6");
+    }
 
-        string source = @"
-            print ""foobar""[10];
-        ";
+    [Fact(Skip = "Not yet supported for UTF8String")]
```

Comment:
I'll go ahead and fix this, but in a separate PR.

Reply:
Ah, my bad. This shouldn't be "fixed" for the time being; it's not something we can easily support. #479